libs/x-content/src/main/java/org/elasticsearch/common/xcontent/XContentType.java

```diff
@@ -114,13 +116,19 @@ public XContent xContent() {
         }
     };
 
+    private static final Pattern COMPATIBLE_API_HEADER_PATTERN = Pattern.compile(
+        "(application|text)/(vnd.elasticsearch\\+)?([^;]+)(\\s*;\\s*compatible-with=(\\d+))?",
```

Comment:
> (vnd.elasticsearch\\+)?

Am I reading this correctly that the `vnd.elasticsearch` part is completely optional? I thought the intent was to only add `compatible-with` when `vnd.elasticsearch` is present?

Reply:
you are right. This is possibly a bit oversimplified.
Do you think we should aim to be more strict on the regex side or should we validate the values after the parsing with more relaxed regex?

The more strict could look like 
```
Pattern.compile(
        "(application|text)/((vnd\\.elasticsearch\\+(json|smile|yaml|cbor)(\\s*;\\s*compatible-with=(\\d+)))" +
            "|(json|smile|yaml|cbor))");
```

there is a simplistic validation in `parseVerion` method, but it still allows to add `compatible-with` even when subtype `vnd.elasticsearch` is not present. The version won't be parsed though